src/chainrulescore.jl

```diff
@@ -0,0 +1,129 @@
+using ChainRulesCore
+using ArrayLayouts: MemoryLayout, AbstractColumnMajor
```

Comment:
Do we want to depend on ArrayLayouts just for this?

I believe something like this ought to work:
```
colmajor(x) = (is_strided(x) && stride(x,1) == 1) ? x : collect(x) 
```

Reply:
`ArrayLayouts` seems pretty minimal and should be more extensible by other packages, so I think it's fine, but I can also use your version, if people are worried about adding dependencies to NNlib.

Reply:
I suspect the entire universe is going to end up depending on both ArrayLayouts.jl and the competing ArrayInterface.jl, directly or indirectly. There was a nice Base PR that would make `strides(A,1) === nothing` if no strides, but it gathers dust on the shelf.

Reply:
Ok, I just left this as-is for now, we can always revisit that in the future.

Reply:
there have been some concerns in the past about too many dependencies here causing slowing down in downstream packages (e.g. CUDA.jl), so let's avoid adding another  dependency  if we could just use a oneliner

Reply:
Ok, I removed the dependency.